collections/duties.js

```diff
@@ -30,4 +33,29 @@ function newDutie(data) {
   });
 }
 
-module.exports.newDutie = newDutie;
\ No newline at end of file
+function getDutie(name, id) {
+  const collection = require("../start").dutiesCollection;
+  const resHandler = require("../start").server.getResHandler();
+  let searchedParams = {};
+
+  if (id && !utils.isMatchesRegex(/[0-9a-fA-F]{24}/, id)) {
+    resHandler.routeNotFound("the id that was sent is not legal, it needs to be of 24 chars in alpha-numeric")
+    return;
+  }
+
+  utils.setParamsForSearch(searchedParams, ["name", "_id"], [name, utils.myObjectId(id)]);
+  
+  collection.find(searchedParams).toArray(function (err, docs) {
+    let isAdditionalParams = (name || id);
+    if (docs.length <= 0 && isAdditionalParams) {
```

Comment:
this elaborate if else structures seems unnecessary. can it not be resolved by checking the type and length of docs? and nothing else?